### ISSUE TITLE
asZnUrl changes relative path to an absolute one when starting with '../'

### ISSUE THREAD

**Issue by alexzandros** [svenvc/zinc#52]
I'm using ZnClient to requests to Wikipedia REST API to get one random article, the response has a 303 status so it starts a redirect wich gets a response code 404, while if I do the same request using Javascript fetch API it works as expected, following the redirect and getting the given resource.

So I started tracing where could the issue be and found the problem is when ZnClient is processing the prepareRedirect message, wich in turn sends `self redirectUrl` whose code follows:

```smalltalk
	| redirectUrl |
	redirectUrl := response location asZnUrl.
	^ redirectUrl isAbsolute
		ifTrue: [ redirectUrl ]
		ifFalse: [ redirectUrl inContextOf: request url ]
```

So after getting to this point I begin testing the `asZnUrl` message in the playground and found that it has problem when the url is a relative path starting with '../' as it converts it to an absolute url as shown: `'../resource/id' asZnUrl   => "/resource/id" ` breaking all the redirecting flow.

**Comment by svenvc** [comment]
Hi,

Thanks for the report. Could you please provide the actual URL and code snippet that shows the problem, please ?

Redirect URLs are fishy, if I remember correctly from years ago. We'll have to look at the details.

Sven

**Comment by alexzandros** [comment]
Hi Sven, follows the snippet showing the problem:

```smalltalk
ZnClient new
	url: 'https://en.wikipedia.org/api/rest_v1/page/random/summary';
	get;
	response. ">>> a ZnResponse(404 Not Found text/html;charset=utf-8 1281B)"
```
Internally it is trying to follow a redirect, since the API response is a 303 code.

Thanks.

**Comment by svenvc** [comment]
Hi Carlos,

The following commit should fix your issue: 001a85c3e917c75e6c9aa20e7b8e03162470f0a8

Could you please confirm ?

Thx (and thanks for the feedback),

Sven

**Comment by alexzandros** [comment]
Works like a charm, thx a lot. Now to keep on learning pharo,

Carlos.
